internal/mode/static/policies/observability/validator.go

```diff
@@ -0,0 +1,162 @@
+package observability
+
+import (
+	"fmt"
+	"slices"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/conditions"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/kinds"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/policies"
+	staticConds "github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/conditions"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/validation"
+)
+
+// Validator validates an ObservabilityPolicy.
+// Implements policies.Validator interface.
+type Validator struct {
+	genericValidator validation.GenericValidator
+}
+
+// NewValidator returns a new instance of Validator.
+func NewValidator(genericValidator validation.GenericValidator) *Validator {
+	return &Validator{genericValidator: genericValidator}
+}
+
+// Validate validates the spec of an ObservabilityPolicy.
+func (v *Validator) Validate(
+	policy policies.Policy,
+	globalSettings *policies.GlobalPolicySettings,
+) []conditions.Condition {
+	obs, ok := policy.(*ngfAPI.ObservabilityPolicy)
+	if !ok {
+		panic(fmt.Sprintf("expected ObservabilityPolicy, got: %T", policy))
+	}
+
+	if globalSettings == nil || !globalSettings.NginxProxyValid {
+		return []conditions.Condition{staticConds.NewPolicyNotAcceptedNginxProxyNotSet()}
```

Comment:
would it make sense to make the policy not valid if telemetry is not enabled?

or perhaps some other reason Accepter/False/<some-other-reason>

this way, the owner of the policy will be aware that it cannot work

Reply:
Yes, good call.